### PR TITLE
Uri vs path

### PR DESCRIPTION
**Pull request by FynnBe**
needs https://github.com/bioimage-io/spec-bioimage-io/pull/288

This PR updates the use of URI classes and pathlib.Path as relative paths are no longer considered valid URIs. Union[URI, Path] is used where appropriate.
Also improves some util functions.

### CODE REVIEW

**Comment by constantinpape** [comment]
Similar to https://github.com/bioimage-io/spec-bioimage-io/pull/288 I can't really review this because the whole Path/URI management is a mess that I don't understand.
But looks like this fixes the tests now, so feel free to merge once you have made all the changes you want.

File: bioimageio/core/build_spec/build_model.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import datetime
 import hashlib
 import os
-from pathlib import Path
+from pathlib import Path, WindowsPath
```
Comment (constantinpape) [inline bioimageio/core/build_spec/build_model.py:4]: WindowsPath is imported here but afaik not used in this file.

Comment (FynnBe) [reply bioimageio/core/build_spec/build_model.py:4]: right, found a nice way (problem was the potential ":" in windows paths interfering with splitting dependency manager and file at ":"). cleaned it up

**Comment by constantinpape** [comment]
Looks like zenodo is having timeout issues so the tests are failing ...

**Comment by FynnBe** [comment]
> Looks like zenodo is having timeout issues so the tests are failing ...

I'd almost merge anyway given that 6b25a3b passes... let's give it another try first

**Comment by constantinpape** [comment]
> I'd almost merge anyway given that [6b25a3b](https://github.com/bioimage-io/core-bioimage-io-python/commit/6b25a3b6c7f3367a73f5737709aff233b2f87e27) passes... let's give it another try first

Good call, now it passes :). I am gonna merge as soon as the conda build test is done.

**Pull request merged**